bravado/client.py

```diff
@@ -215,6 +215,16 @@ def construct_params(self, request, op_kwargs):
             if not remaining_param.required and remaining_param.has_default():
                 marshal_param(remaining_param, None, request)
 
+    def log_warning_for_deprecated_op(self):
+        # TODO: Move this logic to bravado_decorators
```

Comment:
So `deprecated` is part of the Swagger spec, so we're obviously supporting that functionality.

However, `x-deprecated-date` and `x-removal-date` are probably only going to be used within Yelp.  Having to split this functionality out into a separate internal package is probably more hassle than it is worth for a couple of reasons. First, this isn't a core feature. Second, all it does is surface more detailed info in the deprecation warning.

As such, the code should treat the `x-` fields as optional.  If they exist, the deprecation message should be updated accordingly.  Each field stands on its own and does not require the other, I would imagine.